pkg/agent/route/route_linux.go

```diff
@@ -152,7 +152,74 @@ func (c *Client) syncIPInfra() {
 		klog.Errorf("Failed to sync iptables: %v", err)
 		return
 	}
-	klog.V(3).Infof("Successfully synced node iptables")
+	if err := c.syncRoutes(); err != nil {
+		klog.Errorf("Failed to sync routes: %v", err)
+	}
+	if err := c.syncGwIp(); err != nil {
+		klog.Errorf("Failed to sync Ip configuration of gateway: %v", err)
+	}
+	klog.V(3).Infof("Successfully synced node iptables and routes")
+}
+
+func (c *Client) syncRoutes() error {
+	routeList, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
+	if err != nil {
+		return err
+	}
+	routes := []*netlink.Route{}
+	c.nodeRoutes.Range(func(_, v interface{}) bool {
+		for _, route := range v.([]*netlink.Route) {
+			routes = append(routes, route)
+		}
+		return true
+	})
+	for _, route := range routes {
+		exist := false
+		for i := range routeList {
+			if routeEqual(route, &routeList[i]) {
```

Comment:
The nested loop is O(n^2), which may be bad in a cluster of thousands of nodes, especially when there are many other unknown routes configured on the node.
could you optimize it to O(n) by using a map?